### DISCUSSION TITLE
MirrorMaker 2 - Openshift routes

### DISCUSSION THREAD

**Discussion by john-mcpeek** [strimzi/strimzi-kafka-operator#4927] [Q&A]
I was having a problem where topics were not getting replicated between 2 clusters. To reproduce it in a move controlled way I created 2 projects proj1 and proj2. Installed a default kafka cluster in each project with 9094 external route added. Created a mm2 in proj1 with proj2 as the source and proj1 at the target. I did copy the cert from proj2 into a secret in proj1 (remote-kafka-ca-crt) and used that in the tls config.

If I  set the source to use the service (**st2-cluster-kafka-bootstrap.proj2.svc:9093**) everything works.
If I use the route (**st2-cluster-kafka-bootstrap-proj2.apps.blah.oscplatform.site:443**) I get:

```
Metadata update failed due to authentication error (org.apache.kafka.clients.admin.internals.AdminMetadataManager) [kafka-admin-client-thread | adminclient-12]
org.apache.kafka.common.errors.SslAuthenticationException: SSL handshake failed
Caused by: javax.net.ssl.SSLHandshakeException: No subject alternative DNS name matching st2-cluster-kafka-external-bootstrap.proj2.svc found.
```

The mm2 yaml is attached. Is there something I am doing wrong on the route config?
I
[mm2-ssl.txt](https://github.com/strimzi/strimzi-kafka-operator/files/6442688/mm2-ssl.txt)
[kafka-clusters.txt](https://github.com/strimzi/strimzi-kafka-operator/files/6442719/kafka-clusters.txt)

**Comment by scholzj** [comment]
Wait, the error suggests that when it fails when you are connecting to `st2-cluster-kafka-external-bootstrap.proj2.svc` ... so are you sure that `st2-cluster-kafka-external-bootstrap.proj2.svc:9093` works and `st2-cluster-kafka-bootstrap-proj2.apps.blah.oscplatform.site:443` doesn't?

In general, you should be able to disable the TLS hostname validation by setting `ssl.endpoint.identification.algorithm: ""` in the config section of the cluster. But with routes it should normally work.
